src/main/java/org/cru/crs/api/process/PaymentProcessor.java

```
@@ -120,6 +124,71 @@ public void saveNewPayment(Payment payment, CrsApplicationUser loggedInUser)
 		}
 	}
 
+	public void adminUpdatePaymentInfo(Payment newPaymentInfo, CrsApplicationUser loggedInUser)
+	{
+		validatePayment(newPaymentInfo);
+
+		ConferenceEntity conferenceEntity = getConferenceForThisPayment(newPaymentInfo);
+
+		try
+		{
+			// This should only be done by admin
+			verifyUserHasUpdateConferenceRights(conferenceEntity, loggedInUser);
+
+			PaymentEntity paymentToUpdate = paymentService.getPaymentById(newPaymentInfo.getId());
+			if (paymentToUpdate == null)
+			{
+				throw new BadRequestException("Payment: " + newPaymentInfo.getId() + " not found");
+			}
+
+			if (newPaymentInfo.getPaymentType().equals(PaymentType.SCHOLARSHIP))
+			{
+				paymentToUpdate.setTransferSource(newPaymentInfo.getScholarship().getSource());
+				paymentToUpdate.setAmount(newPaymentInfo.getAmount());
+			}
+			else if (newPaymentInfo.getPaymentType().equals(PaymentType.TRANSFER))
+			{
+				paymentToUpdate.setTransferSource(newPaymentInfo.getTransfer().getSource());
+				paymentToUpdate.setAmount(newPaymentInfo.getAmount());
+			}
+			else if (newPaymentInfo.getPaymentType().equals(PaymentType.CASH))
+			{
+				paymentToUpdate.setAmount(newPaymentInfo.getAmount());
+			}
+			else if (newPaymentInfo.getPaymentType().equals(PaymentType.CHECK))
+			{
+				paymentToUpdate.setAmount(newPaymentInfo.getAmount());
+			}
+
+			paymentToUpdate.setDescription(newPaymentInfo.getDescription());
+
+			paymentService.updatePayment(paymentToUpdate, loggedInUser);
+		}
+		catch (Exception e)
+		{
+			// not admin
```

Swallowed exceptions are always bad.  It's not really ideal to expect an exception as a part of normal processing.. That's not what they were intended for.

It would be better to do a check before hand, either in this method or in the calling class, to see if the user is an admin.  If so, then execute the code.  If not, then skip it.